### PR TITLE
Remove "Enable the Visual Studio hosting process" UI...

### PR DESCRIPTION
**Pull request by KevinH-MS**

### CODE REVIEW

**Comment by KevinH-MS** [comment]
FYI @ManishJayaswal @srivatsn @balajikris 
/CC @MattGertz 

This is the work to remove the VS Hosting Process UI from the Debug Property Page.

**Comment by KevinH-MS** [comment]
FYI @jasonmalinowski

**Comment by Pilchie** [comment]
:+1:

File: src/Microsoft.VisualStudio.Editors/PropPages/DebugPropPage.vb [hunk 1]
```diff
@@ -135,13 +132,6 @@ Namespace Microsoft.VisualStudio.Editors.PropertyPages
                 Me.rbStartProgram.Checked = (action = VSLangProj.prjStartAction.prjStartActionProgram)
                 Me.rbStartURL.Checked = (action = VSLangProj.prjStartAction.prjStartActionURL)
                 Me.StartURL.Enabled = (action = VSLangProj.prjStartAction.prjStartActionURL)
-
-                If action = VSLangProj.prjStartAction.prjStartActionProject Then
-                    EnableControl(UseVSHostingProcess, True)
```
Comment (jasonmalinowski) [inline src/Microsoft.VisualStudio.Editors/PropPages/DebugPropPage.vb]: I won't dare ask why EnableControl is used to enable controls but you don't need it to disable controls.

File: src/Microsoft.VisualStudio.Editors/SettingsDesigner/AppConfigSerializer.vb [hunk 2]
```diff
@@ -402,8 +397,8 @@ Namespace Microsoft.VisualStudio.Editors.SettingsDesigner
                 Dim map As New ExeConfigurationFileMap
                 Try
                     map.ExeConfigFilename = AppConfigDocData.Name
-                    map.LocalUserConfigFilename = ConfigHelperService.GetUserConfigurationPath(hierSp, project, ConfigurationUserLevel.PerUserRoamingAndLocal, UnderVsHost, BuildConfig)
-                    map.RoamingUserConfigFilename = ConfigHelperService.GetUserConfigurationPath(hierSp, project, ConfigurationUserLevel.PerUserRoaming, UnderVsHost, BuildConfig)
+                    map.LocalUserConfigFilename = ConfigHelperService.GetUserConfigurationPath(hierSp, project, ConfigurationUserLevel.PerUserRoamingAndLocal, underHostingProcess:=False, buildConfiguration:=BuildConfig)
```
Comment (jasonmalinowski) [inline src/Microsoft.VisualStudio.Editors/SettingsDesigner/AppConfigSerializer.vb]: Will this cause existing projects that already have the flag explicitly set to get cleared?

File: src/Microsoft.VisualStudio.Editors/PropPages/DebugPropPage.vb (hunk 1 above)
Comment (KevinH-MS) [inline src/Microsoft.VisualStudio.Editors/PropPages/DebugPropPage.vb]: I couldn't tell you anyway... :smile:

File: src/Microsoft.VisualStudio.Editors/SettingsDesigner/AppConfigSerializer.vb (hunk 2 above)
Comment (KevinH-MS) [inline src/Microsoft.VisualStudio.Editors/SettingsDesigner/AppConfigSerializer.vb]: Good question--let me try to look.

Comment (KevinH-MS) [inline src/Microsoft.VisualStudio.Editors/SettingsDesigner/AppConfigSerializer.vb]: Okay, as far as I can tell, what the ```underHostingProcess``` parameter does is determine which app.config to read (App.exe.config vs. App.vshost.exe.config).  The value is not actually persisted in said config.

**Pull request merged**